server/services/v1/query_runner.go

```diff
@@ -720,6 +730,100 @@ func (runner *SearchQueryRunner) getFieldSelection(collFields []*schema.Queryabl
 	return factory, nil
 }
 
+type SubscribeQueryRunner struct {
+	*BaseQueryRunner
+
+	req       *api.SubscribeRequest
+	streaming SubscribeStreaming
+}
+
+func (runner *SubscribeQueryRunner) Run(ctx context.Context, tx transaction.Tx, tenant *metadata.Tenant) (*Response, context.Context, error) {
+	db, err := runner.GetDatabase(ctx, tx, tenant, runner.req.GetDb())
+	if ulog.E(err) {
+		return nil, ctx, err
+	}
+
+	collection, err := runner.GetCollections(db, runner.req.GetCollection())
+	if ulog.E(err) {
+		return nil, ctx, err
+	}
+
+	table, err := runner.encoder.EncodeTableName(tenant.GetNamespace(), db, collection)
+	if ulog.E(err) {
+		return nil, ctx, err
+	}
+
+	startTime := time.Now().UTC().Format(time.RFC3339Nano)
+	endTime := time.Now().AddDate(34, 0, 0).Format(time.RFC3339Nano)
```

Comment:
If endTime is very high then the read-range will lock this whole range and any future write will conflict, should we move this to snapshot read?